### ISSUE TITLE
Auto-Matched Color Tome curve doesn't work very well with Fuji's x-trans RAFs

### ISSUE THREAD

**Issue by dpecka** [Beep6581/RawTherapee#6108]
Hello,

first few RAFs for the beginning, you can check them - tested it with stable 5.8 + RawTherapee_dev_5.8-2735-g8bb06c2_20210121

https://infophagia.com/ntz/paste/DSCF2782.tgz - especially fancy shape here
https://infophagia.com/ntz/paste/DSCF2776.tgz
https://infophagia.com/ntz/paste/DSCF2735.tgz - this is image from screenshots below

I've found an issue that FUJIFILM's RAFs don't work well with RT because the initial Color Tone curve is guessed from embedded JPG and result is not very good. Images with broken colours.

After some digging around, I've found, that our competitors software don't have this issue (tested LR and Darktable) and Dartable seems applying a generic brand-like profile .. see please following screenshot

https://infophagia.com/ntz/paste/Screenshot_20210209_105508.png

when RT opens the image, the auto-guessed curve looks like this (see colours in sky washed away)

https://infophagia.com/ntz/paste/Screenshot_20210209_111133.png

based on Dartable I've created a darktable-fuji-like profile which looks like this (please ignore that I used a film-like flavour, I did it only for the proof of concept)

```
> cat ~/.config/RawTherapee/profiles/FUJI-x100f-C.rtc
CatmullRom
0 0
0.05 0.0344645
0.171948 0.17535
0.292675 0.402088
0.462864 0.68235
0.696062 0.936923
1 1
```
and it brought this result:

https://infophagia.com/ntz/paste/Screenshot_20210209_111237.png

Could somebody be please so kind and check what's up with that auto-guessed Tone Color curve shape while working with FUJI RAFs  (RAFs attached)? It seems to me that's clearly broken .. Competitors don't seem to have this issue (tested LR + DT) and in my opinion, a less experienced users with FUJI cameras will be drawn away from using RT while experiencing this.

thanks much and regards, dan

ps. I tested of course a Color Profiles (extracted from Adobe's DNG converter) as described in here 

https://rawpedia.rawtherapee.com/How_to_get_LCP_and_DCP_profiles

but it has nothing to do with shape of guessed Color Tone curve and thus using these "proper & proprietary" profiles has no impact on this issue ..

**Comment by Thanatomanic** [comment]
@dpecka Thanks for reporting here as well, we can take a look.
Could you please deduplicate the text in your post?

**Comment by dpecka** [comment]
Hello,
did (deduplicate), there was accidental paste ... apologize
regards, dan

**Comment by heckflosse** [comment]
Also, https://discuss.pixls.us/t/exposure-tone-curves-profiles-per-brand-or-camera/23367/17

**Comment by Thanatomanic** [comment]
@dpecka Could you please test the Auto-Matched Tone Curve and set the curve to 'Luminance' mode? My guess is that you will be much more pleased with the result.

**Comment by dpecka** [comment]
Hello,

with luminance it looks bit better but if you look a little bit deeper, you can see this

https://infophagia.com/ntz/paste/Screenshot_20210210_233648.png

link on DSCF2782.RAF is above .. I am not a PRO in the field of digital image processing but to be honest, I don't remember that I've ever seen such a shape (that Tone Color curve) before for example with all my Nikon photos (99% of my photography postprocessing are Nikon NEFs)

best regards, dan

ps. I really appreciate very much how you jumped on this issue .. just wow, guys, you're great !!!

**Comment by dpecka** [comment]
Hello,

another prominent example:

https://infophagia.com/ntz/paste/Screenshot_20210219_110650.png
++ RAF here
https://infophagia.com/ntz/paste/DSCF2952.RAF.gz

regards, dan

**Comment by dpecka** [comment]
Hello,

can't we please have some labels on this ?

best regards, dan

ps. I will have during the weekend the outage on my (pastebin) server infophagia.com due to the migration to new VPS provider so won't be possible to download RAFs

**Comment by todd-prior** [comment]
What are you using for a color input profile in CMS...this alters the curve...for example if you have auto match on and then use say a dcp profile and you enable the tone curve in it then the auto match curve becomes almost strait as the toning is the combined effect. I wonder if something in your input profile is creating the extreme curves....

**Comment by Thanatomanic** [comment]
@todd-prior No, this is not the case. There is extensive discussion on this topic already: https://discuss.pixls.us/t/exposure-tone-curves-profiles-per-brand-or-camera/23367
I am convinced the matching that @dpecka wants cannot be achieved without doing per-channel matching. This is something RT is not capable of doing currently, and you can argue about whether that would be correct anyway.

**Comment by Thanatomanic** [comment]
Now that I reread this issue I'm actually not sure what Dan wants. @dpecka, can you elaborate on that? What would be your desired outcome of this report?

Some things to consider:
1. The matching algorithm is actually working as intended. Weird bumps are mostly due to a very limited dynamic range of the raw image and a big difference with the embedded JPEG.
2. Matching is done on the luminance of the images. Setting the tone curve to 'Luminance' will give a better result.
3. If the JPEG processing of the camera is adding a magic sauce (or, in this case, doing matching on independent channels), there is no way to replicate this perfectly in RT.

**Comment by todd-prior** [comment]
NP but just for kicks I tried loading the image...neutral profile, use the dcp file for the camera. Enabled all elements of the DCP...left the automatch using saturation-hue model on and it was a more subdued curve and much closer to the jpg...small bump in saturation and contrast made them quite close. With out the DCP the matches of all curves were not as good

**Comment by todd-prior** [comment]
Certainly the model of the automatch does change the output a lot so that requires exploration. I think you suggested luminance but that made the image have weird blues in the highlights….saturation hue I found to be closer…..case in point a few tweaks and adjustments are still required for the best match which might not be super close in all cases.

On another note. I thought just hitting autolevels gave a nicer rendition of the image…perhaps a bit too much exposure in the highlights but the jpg is murky and dark without some help …I understand that was not the original query….

I think with some experimentation it would be possible to get a starting point but I think there will always be one or two tweaks…..

**Comment by dpecka** [comment]
I've opened this report because I thought that the behaviour is buggy ... my apologize if I misinterpreted it .. I have Nikon cameras for long years and in past 3 years I've been using RawTherapee which works and worked 100% good with Nikon NEFs, even with underexposed .. then I bought month ago a FUJIFILM X100F camera and suddenly 66% of (yes, 2/3) of images while opened in RT looked broken ..

I've learned a lot in past days and also thanks to thread at pixls.us so I can now do corrective steps that RAFs opened in RT look now similar like JPG preview ..

I don't know if this is bug but the reality is, that in general the images from this Fujifilm camera opened in RT looks bad and without specific skill and knowledge they have improper Color Tone curve ...
